components/devtools/actors/root.rs

```diff
@@ -220,13 +232,30 @@ impl Actor for RootActor {
             },
 
             "getTab" => {
-                let tab = registry.find::<TabDescriptorActor>(&self.tabs[0]);
-                let reply = GetTabReply {
-                    from: self.name(),
-                    tab: tab.encodable(registry),
-                };
-                let _ = stream.write_json_packet(&reply);
-                ActorMessageStatus::Processed
+                if let Some(serde_json::Value::Number(browserId)) = msg.get("browserId") {
+                    let targetTab = self
+                        .tabs
+                        .iter()
+                        .map(|target| {
+                            registry
+                                .find::<TabDescriptorActor>(target)
+                                .encodable(registry, true)
```

Comment:
Why encode here, rather than when we find the actual tab we want to send back?

Reply:
I did it this way since `TabDescriptorActorMsg` contains the `broswerId`, but `TabDescriptorActor` doesn't, and we need to filter them by that, I thought of doing the conversion beforehand, but I can absolutely try to do it other way.

Reply:
Ah, I see. That's reasonable!